### DISCUSSION TITLE
When did TypeChain refactor "commons.ts" to "common.d.ts"?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-ts/TypeChain#526] [Q&A] [answered]
In commit [411805d](https://github.com/paulrberg/hardhat-packager/commit/411805dd64b7d6e06236ebccc26b5ee23806e701) of [hardhat-packager](https://github.com/paulrberg/hardhat-packager), I added `common` as the name of the "excluded files" (excluded here means it is not deleted). Before that I was only excluding `commons`.

The question is when does Hardhat generate the `common.d.ts` file, and when does it generate `commons.ts`? Or, is it that you have switched to only generating `common.d.ts`?

**Comment by hasparus** [comment, marked as answer]
Hey @paulrberg.

We changed from _common**s**.ts_ to _common.**d**.ts_ in #472 and the change was released in `@typechain/ethers-v5@7.1.1`.  We missed it in changesets 😓

At this moment, TypeChain generates only _common.**d**.ts_.

---

What may add to the confusion, I switched from _common.**d**.ts_ to _common.ts_ in https://github.com/dethcrypto/TypeChain/pull/504 which is not released yet. However, looking at https://github.com/paulrberg/hardhat-packager/commit/411805dd64b7d6e06236ebccc26b5ee23806e701 it seems it won't be a problem for you.

**Reply by PaulRBerg** [reply]
Thanks for explaining! Indeed the rename to _common.ts_ won't be a problem for the distributed package, but it will be for my tests: https://github.com/paulrberg/hardhat-packager/blob/67b426826bb486e98247392841490d6d75d79b83/test/packager.test.ts#L91.

**Reply by krzkaczor** [reply]
@paulrberg yes, sorry for that. I think we assumed that it's not part of the public interface. Which was a mistake... 

We are going to release new major maybe today.

Paul it was great meeting you in person in Lisbon ;)

**Reply by PaulRBerg** [reply]
Don't worry, @krzkaczor! It's a minor thing.

Indeed it was great to catch up in person 🙌

**Comment by PaulRBerg** [comment]
Hey, I just upgraded to `typechain@7.0.0` and `@typechain/ethers-v5@9.0.0` and it looks like you have switched back to using `common.ts`.

Can you confirm this was intentional? Or was it an accidental reversion?

**Reply by PaulRBerg** [reply]
Hey @krzkaczor or @hasparus  sorry to tag you guys but this is a real blocker.

Any chance you could take a look at my question? 🙏

**Reply by hasparus** [reply]
Hey @paulrberg, Forgive me, I must have missed this in GitHub notifications.

Yes, we did switch to _common.ts_ in #504. I mentioned this in my previous comment in this discussion https://github.com/dethcrypto/TypeChain/discussions/526#discussioncomment-1539236. I'm sorry for the trouble. We'll try not to touch filename this anymore.

**Reply by PaulRBerg** [reply]
No worries! Thanks for confirming. I missed the last part of your other comment.
